compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1508,12 +1508,15 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         })
     }
 
+    /// Return Some(true) if the obligation's predicate type applies to the env_predicate, and
+    /// Some(false) if it does not. Returns None in the case that the projection type is a GAT,
+    /// and applying this env_predicate constrains any of the obligation's GAT substitutions.
     pub(super) fn match_projection_projections(
```

Comment:
Comment is slightly outdated now that the return type has changed to an enum

Reply:
Good catch.